pandas/core/generic.py

```diff
@@ -4529,11 +4532,11 @@ def sort_index(
         new_data = self._mgr.take(indexer, axis=baxis, verify=False)
 
         # reconstruct axis if needed
-        new_data.axes[baxis] = new_data.axes[baxis]._sort_levels_monotonic()
+        new_data.set_axis(baxis, new_data.axes[baxis]._sort_levels_monotonic())
 
         if ignore_index:
             axis = 1 if isinstance(self, ABCDataFrame) else 0
-            new_data.axes[axis] = ibase.default_index(len(indexer))
+            new_data.set_axis(axis, ibase.default_index(len(indexer)))
```

Comment:
can you comment on why these need to be changed?

Reply:
In hindsight, it's probably not needed anymore. Originally, I changed the `axes` attribute on ArrayManager to be (index, columns) order. But afterwards, I changed this to be stored in `_axes`, and having an `axes` property that switches the internal order, to match the "public" interface of BlockManager. 

So with that change, I could revert those edits (I *think*, but can check that)

Reply:
Just checked this again, and so this change is needed (without it I get failures for the array manager). 
The reason is that `mgr.axes` is no longer the actual "storage" of the axes for the ArrayManager, but is there stored in `_axes`. And because `mgr.axes` returns a list, it's actually updating the list in place, and not the actual storage. Which is a bit of a gotcha now with the manager's `axes` attribute.

Reply:
thanks for explaining.  we should make another attempt to get axes out of FooManager before too long